src/RabbitMQPubSubServiceProvider.php

```diff
@@ -20,12 +25,88 @@ public function boot()
 
     public function register()
     {
-        App::singleton(RabbitMQPubSub::class, function (Application $app) {
-            return new RabbitMQPubSub($app, $app->make(RabbitMQConnectionManager::class));
+        $this->app->singleton(PubSub::class, function () {
+            /** @var ConnectionInterface $connection */
+            $connection = $this->app->make(ConnectionInterface::class);
+            $pubSub = new PubSub($connection);
+
+            $consumers = config('rabbitMQPubSub.consumers')
+                ?? throw MissingConfigException::create('rabbitMQPubSub.consumers');
+
+            $declaredQueues = [];
+
+            foreach ($consumers as $topicName => $topicInfo) {
```

Comment:
subscribers should only be created when `laravel-rabbitmq-pubsub:consumer` is run

Reply:
Fixed now